src/modules/flow/http-server/http-server.c

```diff
@@ -214,79 +193,100 @@ boolean_close(struct sol_flow_node *node, void *data)
-
 static int
-string_response_cb(void *data, struct sol_http_request *request)
+common_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
+    const struct sol_flow_packet *packet)
 {
-    int r = 0;
-    uint16_t idx;
-    bool send_json = false;
-    enum sol_http_method method;
     struct http_data *mdata = data;
-    char *response_str;
-    struct sol_http_param_value *value;
-    struct sol_http_response response = {
-        .api_version = SOL_HTTP_RESPONSE_API_VERSION,
-        .content = SOL_BUFFER_INIT_EMPTY,
-        .param = SOL_HTTP_REQUEST_PARAM_INIT,
-        .response_code = SOL_HTTP_STATUS_OK
-    };
+    const struct http_server_node_type *type;
+    int r;
 
-    method = sol_http_request_get_method(request);
-    response.url = sol_http_request_get_url(request);
+    type = (const struct http_server_node_type *)
+        sol_flow_node_get_type(node);
 
-    SOL_HTTP_PARAM_FOREACH_IDX (sol_http_request_get_params(request), value, idx) {
-        switch (value->type) {
-        case SOL_HTTP_PARAM_POST_FIELD:
-            if (streq(value->value.key_value.key, "value")) {
-                free(mdata->value.s);
-                mdata->value.s = strdup(value->value.key_value.value);
-                if (!mdata->value.s) {
-                    r = -ENOMEM;
-                    SOL_WRN("Could not duplicate the string");
-                    goto end;
-                }
-
-                r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
-                SOL_INT_CHECK_GOTO(r, < 0, end);
-            }
-            break;
-        case SOL_HTTP_PARAM_HEADER:
-            if (streq(value->value.key_value.key, HTTP_HEADER_ACCEPT)) {
-                if (strstr(value->value.key_value.value, HTTP_HEADER_CONTENT_TYPE_JSON))
-                    send_json = true;
-            }
-            break;
-        default:
-            break;
-        }
+    r = type->process_cb(mdata, packet);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+boolean_post_cb(struct http_data *mdata, struct sol_flow_node *node,
+    struct sol_http_param_value *value)
+{
+    if (streq(value->value.key_value.value, "true"))
+        mdata->value.b = true;
+    else if (streq(value->value.key_value.value, "false"))
+        mdata->value.b = false;
+    else
+        return -EINVAL;
+
+    sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_BOOLEAN__OUT__OUT,
+            mdata->value.b);
+
+    return 0;
+}
+
+static int
+boolean_response_cb(struct http_data *mdata, struct sol_buffer *content, bool json)
+{
+    int r;
+    char response_str[128];
+
+    if (json) {
+        r = snprintf(response_str, sizeof(response_str), "{\"%s\": %s}", mdata->path,
+            mdata->value.b == true ? "true" : "false");
```

Comment:
this won't work later to convert to group, since if you call this function for each group element, the result will be `{"/path": {"/path": true}}` which is weird. Maybe have another virtual to just append the JSON value, then this function would call it:
```c
if (json) {
    sol_buffer_reset(content);
    r = sol_buffer_append_printf(content, "{\"%s\":", mdata->path);
   SOL_INT_CHECK(r, < 0, r);
   r = type->json_append_value(mdata, content);
   SOL_INT_CHECK(r, < 0, r);
   r = sol_buffer_append_slice(content, sol_str_slice_from_str("}"));
   SOL_INT_CHECK(r, < 0, r);
}   
```

this also shows that you should be using `sol_buffer_append_printf()` instead of `snprintf()` + `sol_buffer_set_slice()`.